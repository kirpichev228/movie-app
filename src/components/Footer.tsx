import React from 'react'

const Footer = () => {
  return (
    <footer className='h-[10vh] 
    text-white 
      flex 
      justify-between 
      items-center 
      py-5 
      px-20 
      w-full 
      bg-[#17161b] 
      border-t-2 
      border-[#1b1a1f]'>
        <span>
            2022
        </span>
        <a 
            href="https://github.com/kirpichev228/movie-app/tree/dev" 
            target='_blank'
            className='hover:text-[#876afe] transition-all' 
            rel="noreferrer"
        > 
            GitHub 
        </a>
    </footer>
  )
}

export default Footer