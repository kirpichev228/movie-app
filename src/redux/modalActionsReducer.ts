const defaultState = {
    modalOpen: false,
    content: ''
}

export const modalActionsReducer = (state = defaultState, action: reducerActionType) => {
    switch(action.type) {

        case ModalEnum.open:
            return {...state, modalOpen: action.payload}

        case ModalEnum.setContent:
            return {...state, content: action.payload}

        default:
            return state
    }
}