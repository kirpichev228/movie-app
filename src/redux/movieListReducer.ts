const defaultState = {
    itemsList: [],
    itemsListCopy: [],
}

export const movieListReducer = (state = defaultState, action: reducerActionType) => {
    switch (action.type) {
        
        case ListFilterEnum.set:
            return {...state, itemsList: state.itemsList = action.payload}
        
        case ListFilterEnum.search:
            return {...state, itemsList: action.payload}

        case ListFilterEnum.setCopy:
            return {...state, itemsListCopy: state.itemsListCopy = action.payload}
            
        case ListFilterEnum.edit:
            return {...state, itemsList: state.itemsListCopy[action.payload] = action.value}
        
        case ListFilterEnum.add:
            return {...state, itemsList: state.itemsListCopy.unshift(action.payload) }

        case ListFilterEnum.delete: 
            return {...state, itemsList: state.itemsListCopy.splice(action.payload, 1)}

        default:
            return state
    }
}