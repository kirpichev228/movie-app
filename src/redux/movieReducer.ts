const defaultState = {
    item: {
        budget: 0,
        genres: [''],
        id: 0,
        overview: 'ой',
        poster_path: '',
        release_date: 'ой',
        revenue: 322,
        runtime: 228,
        tagline: 'ой',
        title: 'ой',
        vote_average: 8081,
        vote_count: 0,
      }
}

export const movieReducer = (state = defaultState, action: reducerActionType) => {
    switch (action.type) {

        case MovieAsctionEnum.pick:
            return {...state, item: state.item = action.payload}

        case MovieAsctionEnum.delete:
            return {...state, item: state.item = state.itemsList[action.payload]}
            
        case MovieAsctionEnum.edit:
            return {...state, item: state.item = action.payload}
                
        default:
            return state
    }
}