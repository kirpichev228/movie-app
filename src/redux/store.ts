import { applyMiddleware, createStore } from "redux"
import thunk from 'redux-thunk';
import { MovieAsctionEnum, IStoreItem, reducerActionType, ListFilterEnum, ModalEnum, WatchListEnum, GenreSort } from "../models"

const defaultState: IStoreItem = {
    item: {
      budget: 0,
      genres: [''],
      id: 0,
      overview: 'ой',
      poster_path: '',
      release_date: 'ой',
      revenue: 322,
      runtime: 228,
      tagline: 'ой',
      title: 'ой',
      vote_average: 8081,
      vote_count: 0,
    },
    itemsList: [],
    itemsListCopy: [],
    modalOpen: false,
    content: '',
    deleted: false,
    watchList: [],
    selectedGenres: [],
    pagination: 1,
    pageItems: [],
    pageStartIndex: 0,
    pageEndIndex: 24,
}


const reducer = (state = defaultState, action: reducerActionType) => {
    switch(action.type) {

        case MovieAsctionEnum.PICK_MOVIE:
            return {...state, item: state.item = action.payload}

        case MovieAsctionEnum.DELETE_MOVIE:
            return {...state, item: state.item = state.itemsList[action.payload]}
            
        case MovieAsctionEnum.EDIT_MOVIE:
            return {...state, item: state.item = action.payload}
    
        case ListFilterEnum.SET:
            return {...state, itemsList: state.itemsList = action.payload}
        
        case ListFilterEnum.SEARCH:
            return {...state, itemsList: action.payload}

        case ListFilterEnum.SET_PAGE:
            return {...state, pagination: state.pagination = action.payload}

        case ListFilterEnum.SET_PAGE_ITEMS:
            return {...state, pageItems: state.pageItems = action.payload}

        case ListFilterEnum.SET_COPY:
            return {...state, itemsListCopy: state.itemsListCopy = action.payload}

        case ListFilterEnum.SET_START:
            return {...state, pageStartIndex: state.pageStartIndex = action.payload}

        case ListFilterEnum.SET_END:
            return {...state, pageEndIndex: state.pageEndIndex = action.payload}
            
        case ListFilterEnum.EDIT_LIST:
            return {...state, itemsList: state.itemsListCopy[action.payload] = action.value}
        
        case ListFilterEnum.ADD_MOVIES:
            return {...state, itemsList: state.itemsListCopy.unshift(action.payload) }

        case ListFilterEnum.DELETE_MOVIES: 
            return {...state, itemsList: state.itemsListCopy.splice(action.payload, 1)}
        
        case ModalEnum.OPEN_MODAL:
            return {...state, modalOpen: action.payload}

        case ModalEnum.SET_CONTENT:
            return {...state, content: action.payload}

        case ModalEnum.DELETE_CHECK:
            return {...state, deleted: action.payload}

        case WatchListEnum.PUSH_TO_LIST:
            return {...state, watchList: state.watchList.concat(action.movie)}
        
        case WatchListEnum.DELETE_FROM_LIST:
            return {...state, watchList: state.watchList = action.movie }

        case WatchListEnum.CLEAR_LIST:
            return {...state, watchList: state.watchList = []}

        case GenreSort.ADD_GENRE:
            return {...state, selectedGenres: state.selectedGenres.concat(action.genre) }

        case GenreSort.DELETE_GENRE:
            return {...state, selectedGenres: state.selectedGenres.filter(item => item !== action.genre) }

        default:
            return state
    }
}

export const store = createStore(reducer, applyMiddleware(thunk))
