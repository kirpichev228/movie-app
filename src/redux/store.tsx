import { stat } from "fs/promises";
import { applyMiddleware, createStore } from "redux"
import thunk from 'redux-thunk';
import { MovieAsctionEnum, IStoreItem, reducerActionType, ListFilterEnum } from "../models"

const defaultState: IStoreItem = {
    item: {
      budget: 0,
      genres: [''],
      id: 0,
      overview: 'ой',
      poster_path: '',
      release_date: 'ой',
      revenue: 322,
      runtime: 228,
      tagline: 'ой',
      title: 'ой',
      vote_average: 8081,
      vote_count: 0,
    },
    itemsList: [],
    itemsListCopy: []
}



const reducer = (state = defaultState, action: reducerActionType) => {
    switch(action.type) {

        case MovieAsctionEnum.pick:
            return {...state, item: state.item = action.payload}

        case MovieAsctionEnum.delete:
            return {...state, item: state.item = state.itemsList[action.payload]}
            
        case MovieAsctionEnum.edit:
            return {...state, item: state.item = action.payload}
    
        case ListFilterEnum.set:
            return {...state, itemsList: state.itemsList = action.payload}
        
        case ListFilterEnum.search:
            return {...state, itemsList: action.payload}

        case ListFilterEnum.setCopy:
            return {...state, itemsListCopy: state.itemsListCopy = action.payload}
            
        case ListFilterEnum.edit:
            return {...state, itemsList: state.itemsListCopy[action.payload] = action.value}
        
        case ListFilterEnum.add:
            return {...state, itemsList: state.itemsListCopy.unshift(action.payload) }

        case ListFilterEnum.delete: 
            return {...state, itemsList: state.itemsListCopy.splice(action.payload, 1)}

        default:
            return state
    }
}

export const store = createStore(reducer, applyMiddleware(thunk))
