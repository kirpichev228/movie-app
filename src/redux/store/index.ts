import { createStore, applyMiddleware, combineReducers } from "redux";
import thunk from "redux-thunk";
import { modalActionsReducer } from "../modalActionsReducer";
import { movieListReducer } from "../movieListReducer";
import { movieReducer } from "../movieReducer";

const rootReducer = combineReducers({
    movie: movieReducer,
    list: movieListReducer,
    modal: modalActionsReducer,
})

export const store = createStore(rootReducer, applyMiddleware(thunk))