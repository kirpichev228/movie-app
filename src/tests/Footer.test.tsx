import React from 'react';
import { render, screen } from '@testing-library/react';

import Footer from '../components/Footer'
import '@testing-library/jest-dom'
import { Provider } from 'react-redux';
import { store } from '../redux/store';

describe('FOOTER TESTS', () => {
    test('footer year', () => {
        render(
            <Provider store={store}>
                <Footer/>
            </Provider>
        );
        const footerYear = screen.getByText(/2022/i)
        expect(footerYear).toBeInTheDocument()
        expect(footerYear).toMatchSnapshot()
    })

    test('footer GHlink', () => {
        render(
            <Provider store={store}>
                <Footer/>
            </Provider>
        );
        const footerLinkText = screen.getByText('GitHub')
        expect(footerLinkText).toBeInTheDocument()
        expect(footerLinkText).toMatchSnapshot()
    })
    
})



