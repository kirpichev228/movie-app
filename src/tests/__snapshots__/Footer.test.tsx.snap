// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FOOTER TESTS footer GHlink 1`] = `
<a
  class="hover:text-[#876afe] transition-all"
  href="https://github.com/kirpichev228/movie-app/tree/dev"
  rel="noreferrer"
  target="_blank"
>
  GitHub
</a>
`;

exports[`FOOTER TESTS footer year 1`] = `
<span>
  2022
</span>
`;
